pkg/registry/core/pod/strategy_test.go

```diff
@@ -1671,3 +1671,362 @@ func TestNodeInclusionPolicyEnablementInUpdating(t *testing.T) {
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:               "matchLabelKeys is ignored if the labelSelector is nil",
+			featureGateEnabled: true,
+			pod: &api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"country": "Japan",
+						"city":    "Kyoto",
+					},
+				},
+				Spec: api.PodSpec{
+					Affinity: &api.Affinity{
+						PodAffinity: &api.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+								{
+									MatchLabelKeys:    []string{"country"},
+									MismatchLabelKeys: []string{"city"},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantPod: &api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"country": "Japan",
+						"city":    "Kyoto",
+					},
+				},
+				Spec: api.PodSpec{
+					Affinity: &api.Affinity{
+						PodAffinity: &api.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+								{
+									MatchLabelKeys:    []string{"country"},
+									MismatchLabelKeys: []string{"city"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "the feature gate is disabled and matchLabelKeys is ignored",
+			pod: &api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"country": "Japan",
+						"city":    "Kyoto",
+					},
+				},
+				Spec: api.PodSpec{
+					Affinity: &api.Affinity{
+						PodAffinity: &api.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"region": "Asia",
+										},
+									},
+									MatchLabelKeys:    []string{"country"},
+									MismatchLabelKeys: []string{"city"},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantPod: &api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"country": "Japan",
+						"city":    "Kyoto",
+					},
+				},
+				Spec: api.PodSpec{
+					Affinity: &api.Affinity{
+						PodAffinity: &api.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"region": "Asia",
+										},
+									},
+									MatchLabelKeys:    []string{"country"},
+									MismatchLabelKeys: []string{"city"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.MatchLabelKeysInPodAffinity, tc.featureGateEnabled)()
+
+			pod := tc.pod
+			mutatePodAffinity(pod)
+			if diff := cmp.Diff(tc.wantPod.Spec.Affinity, pod.Spec.Affinity); diff != "" {
+				t.Errorf("%s unexpected affinity (-want, +got): %s", tc.name, diff)
```

Comment:
tc.name is already shown up in L2022 when running the sub-test:
```suggestion
				t.Errorf("unexpected affinity (-want, +got): %s\n", diff)
```